### ISSUE TITLE
Stuck on warming up

### ISSUE THREAD

**Issue by enricofromrome** [GoogleChrome/lighthouse#13273]
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/master/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/master/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://www.enricofromrome.eu/edb6/

### What happened?

Nothing happened: that is the problem! Stuck on warming up. It used to work in the past.

### What did you expect?

Normal behavior

### What have you tried?

Repeated day after. Incognito and not incognito window. Tagging different options.

### How were you running Lighthouse?

Chrome DevTools

### Lighthouse Version

Sorry, I could not reach the report.

### Chrome Version

Version 95.0.4638.54 (Official Build) (64-bit)

**Comment by aDorofeev** [comment]
Reproduces on Chrome 95.0.4638.69 (Official Build) (64-bit) / Lighthouse 8.4.0.

It works fine on some pages:
- [Github](https://github.com/GoogleChrome/lighthouse/issues/13273)
- [Wikipedia](https://en.wikipedia.org/wiki/Main_Page)
- [Facebook news feed](https://www.facebook.com/)

And gets stuck on the "warming up" on others:
- https://www.reddit.com/
- https://new.faberlic.com/

Based on my statistic - it usally works on higher quality pages, and gets stuck on heavier pages, or the ones with lots of errors in js console.

I also face exact same issue in Chrome 97.0.4682.3 (Official Build) dev (64-bit)

**Comment by connorjclark** [comment]
Repro on canary 97.0.4689.0 https://www.enricofromrome.eu/edb6/
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'filter')
    at b.hasImportantResourcesNotCleared (lighthouse.js:1:8169)
```
https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/panels/lighthouse/LighthouseController.ts;l=247;drc=11936d8b107dcc6e311cf0406d6fb7854795bdfd

**Comment by connorjclark** [comment]
tthat "port close is not a function" errors is suspicious but it should be harmless, given how it is handled: https://github.com/GoogleChrome/lighthouse/blob/a48d6166df82ce5e365d507b5b4f03f5a233cc69/lighthouse-core/gather/gather-runner.js#L188

this class should define a close method to fix https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/entrypoints/lighthouse_worker/LighthouseService.ts;l=18?q=setUpWorkerConnection&ss=chromium

**Comment by connorjclark** [comment]
we should wrap the webappmanifest base artifact function in the equivalent error handling that our main artifacts get https://github.com/GoogleChrome/lighthouse/blob/a48d6166df82ce5e365d507b5b4f03f5a233cc69/lighthouse-core/gather/gatherers/web-app-manifest.js#L81 . This should prevent the run from exiting fatally.

Although, I'd expect even a fatal run to not hang forever. We catch and rethrow fatal errors here: https://github.com/GoogleChrome/lighthouse/blob/a48d6166df82ce5e365d507b5b4f03f5a233cc69/lighthouse-core/runner.js#L183

which the lighthouse worker should catch here: https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/entrypoints/lighthouse_worker/LighthouseService.ts;l=64;drc=59f68132b1b2d61464d39b050fd3ff157775fa7d

but... we don't use `await` [here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/entrypoints/lighthouse_worker/LighthouseService.ts;l=62;drc=59f68132b1b2d61464d39b050fd3ff157775fa7d) so it doesn't work :) 

-----
tldr; we should fix two errors here, 1) don't allow base artifacts protocol timeouts to create fatal error 2) correctly recover from a fatal error in lighthouse worker

the root cause of the protocol timeout for "get installability errors" is still at-large.

**Comment by enricofromrome** [comment]
My impression is that it has to do with service worker and installation (no
problem with other similar sites without service worker) . Please notice
that on chrome it works and no errors appear in the console, in the past it
used to work on lighthouse with a good result.

Il mer 3 nov 2021, 01:57 Connor Clark ***@***.***> ha scritto:

**Comment by connorjclark** [comment]
Even with the above CL landed, LH still hangs at this point:

https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/core/sdk/TargetManager.ts;l=81;drc=f290aeebbcf4f9e5662a184a9c6afc964f4a602a

Debugging shows that the target for the service worker is never resumed:

![image](https://user-images.githubusercontent.com/4071474/140189104-db1ec44b-1d02-44c9-b3d2-aa5c4b993714.png)

**Comment by hugues-koolsol** [comment]
On https://todo.koolsol.app, 
lighthouse ends correctly only if I uncheck "vider l'espace de stockage" ( ~empty storage space )
I hope it helps !

URL : https://todo.koolsol.app/
Heure d'exécution : 5 nov. 2021, 13:12 UTC+1
Appareil : Émulation (Moto G4)
Limitation de bande passante réseau : 150 ms TCP RTT, 1 638,4 Kbps throughput (Simulated)
Limitation du processeur : 4x slowdown (Simulated)
Version : devtools
User-agent (hôte) : Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.0 Safari/537.36
User-agent (réseau) : Mozilla/5.0 (Linux; Android 7.0; Moto G (4)) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4590.2 Mobile Safari/537.36 Chrome-Lighthouse
Puissance processeur/mémoire : 1740
Version d'axe : 4.2.3
Generated by Lighthouse 8.5.0

**Comment by enricofromrome** [comment]
The PWA uses indexeddb

Il ven 5 nov 2021, 13:30 Hugues Koolsol ***@***.***> ha
scritto:

**Comment by hugues-koolsol** [comment]
Actually, it uses only localstorage api and not indexedDb.
But you are right if you consider that localstorage looks like indexedDb.

**Comment by enricofromrome** [comment]
It uses both local storage and indexeddb through a service worker, it does
it on start. I am afraid this is a lighthouse problem because the program
keeps working on chrome (android and Linux) and Firefox.

Il sab 6 nov 2021, 12:31 Hugues Koolsol ***@***.***> ha
scritto:

**Comment by enricofromrome** [comment]
At this address you can find an older release:
https://www.enricofromrome.eu/edb2/legmen.htm

The older release used only localstorage and doesn't get stuck on
lighthouse. The new release also has service worker, manifest and indexeddb.



Il giorno dom 7 nov 2021 alle ore 01:08 enrico de bernart <
***@***.***> ha scritto:

**Comment by connorjclark** [comment]
Tracking this in #13396, please follow there.
